pkg/cmd/experimental/syncgroups/sync-groups.go

```diff
@@ -0,0 +1,356 @@
+package syncgroups
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api/latest"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/util"
+	kerrs "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/schema1"
+	syncutil "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/util"
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/api/validation"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	SyncGroupsRecommendedName = "sync-groups"
+
+	syncGroupsLong = `
+Sync OpenShift Groups with records from an external provider.
+
+In order to sync OpenShift Group records with those from an external provider, determine which Groups you wish
+to sync and where their records live. For instance, all or some groups may be selected from the current Groups
+stored in OpenShift, or similarly all or some groups may be selected from those stored on an LDAP server. The
+path to a sync configuration file is required in order to describe how data is requested from the external 
+record store and migrated to OpenShift records. Default behavior is to sync all groups from the LDAP server 
+returned by the LDAP query templates.`
+	syncGroupsExamples = `  // Sync all groups from an LDAP server
+  $ %[1]s --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync specific groups specified in a whitelist file with an LDAP server 
+  $ %[1]s --whitelist=/path/to/whitelist.txt --sync-config=/path/to/sync-config.yaml
+
+  // Sync all OpenShift Groups that have already been synced with an LDAP server
+  $ %[1]s openshift --sync-config=/path/to/ldap-sync-config.yaml
```

Comment:
instead of an `openshift` arg, maybe `--existing-groups` or some other flag indicating that we're resyncing what's already sync'ed, but nothing new.